File: docs/math/code/continued-fraction/compare.py
# CONTINUED FRACTION COMPARISON.
# Expand [..., n] to [..., n-1, 1] if needed.
def expand(a):
    if a[-1] != 1 or len(a) == 1:
        a[-1] -= 1
        a.append(1)
    return a


# Check if a is smaller than b.
def less_than(a, b):
    a = expand(a)
    b = expand(b)
    a = [(-1) ** i * a[i] for i in range(len(a))]
    b = [(-1) ** i * b[i] for i in range(len(b))]
    return a < b


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        input()
        a = list(map(int, input().split()))
        b = list(map(int, input().split()))
        print(int(less_than(a, b)), int(less_than(b, a)))

File: math/code/continued-fraction/compare.cpp
#include <algorithm>
#include <iostream>
#include <tuple>
#include <vector>

// --8<-- [start:core]
// Expand [..., n] to [..., n-1, 1] if needed.
void expand(std::vector<int>& a) {
  if (a.size() == 1 || a.back() > 1) {
    --a.back();
    a.push_back(1);
  }
}

// Check if a is smaller than b.
bool less_than(std::vector<int> a, std::vector<int> b) {
  expand(a);
  expand(b);
  for (int i = 0; i < a.size() - 1 || i < b.size() - 1; ++i) {
    int d = (i < a.size() - 1 ? a[i] : 0) - (i < b.size() - 1 ? b[i] : 0);
    if (i & 1) d = -d;
    if (d < 0) {
      return true;
    } else if (d > 0) {
      return false;
    }
  }
  return false;
}

// --8<-- [end:core]
int main() {
  int t;
  std::cin >> t;
  for (; t; --t) {
    int na, nb;
    std::cin >> na >> nb;
    std::vector<int> a(na), b(nb);
    for (int& x : a) std::cin >> x;
    for (int& x : b) std::cin >> x;
    std::cout << less_than(a, b) << ' ' << less_than(b, a) << std::endl;
  }
  return 0;
}
